xcdat/dataset.py

```diff
@@ -325,114 +317,135 @@ def decode_time_units(dataset: xr.Dataset):
         axis:           T
         long_name:      time
         standard_name:  time
+        calendar:       noleap
 
-    View time coordinate encoding information:
+    View time encoding information:
 
-    >>> ds.time.encoding
+    >>> ds_decoded.time.encoding
     {'source': None, 'dtype': dtype('int64'), 'original_shape': (3,), 'units':
-    'years since 2000-01-01', 'calendar': 'proleptic_gregorian'}
+    'years since 2000-01-01', 'calendar': 'noleap'}
     """
-    time = dataset["time"]
+    time = dataset.cf["T"]
+    time_bounds = dataset.get(time.attrs.get("bounds"), None)
     units_attr = time.attrs.get("units")
-
-    if units_attr is None:
-        raise KeyError(
-            "No 'units' attribute found for time coordinate. Make sure to open "
-            "the dataset with `decode_times=False`."
+    units, ref_date = _split_time_units_attr(units_attr)
+    ref_date = pd.to_datetime(ref_date)
+
+    data = [ref_date + pd.DateOffset(**{units: offset}) for offset in time.data]
```

Comment:
I to find a solution for vectorizing converting `time.data` (raw numerically encoded time coordinates) into `pd.DateOffset` and adding it to `ref_date`, but did not come up with anything. `pd.DateOffset` seems to not support vectorization operations that well.

Performance Metrics with For Loop:
```python
# 115 ms ± 29 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
%timeit -n 10 -p 4  [ref_date + pd.DateOffset(**{units: offset}) for offset in time.data]
```

Performance Metrics using Pandas Series and Map (multiple steps):
```python
#%% Step 0: Read numpy array to Pandas Series
# 61.5 µs ± 4.62 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
%timeit pd.Series(time.data)

#%% Step 1: Update to offset
# 24.1 ms ± 5.74 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
dates_pd = pd.Series(time.data)
%timeit -n 10 dates_pd.map(lambda x: pd.DateOffset(**{units: x}), na_action="ignore")
#%% Step 2: Update to timestamp
# 54.7 ms ± 4.66 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
dates_pd1  = dates_pd.map(lambda num: pd.DateOffset(**{units: num}), na_action="ignore")
%timeit -n 10 dates_pd1.map(lambda offset: ref + offset, na_action="ignore")
```

Performance Metrics using Pandas Series and Map (single liner):
```python
# 132 ms ± 28.7 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
# %timeit -n 10 dates_pd.map(lambda offset: reference_date+pd.DateOffset(**{units:offset}), na_action="ignore")
```

Reply:
The for loop operation is only slightly slower than pandas and map with multiple steps and on-par with the pandas single liner. 

I wasn't able to find a vectorized numpy solution.